FlatBuffersSwift/FlatBuffersBuilder.swift

```diff
@@ -100,69 +126,108 @@ public final class FBBuilder {
+    /**
+     Append a scalar value to the buffer
+     
+     - parameters:
+         - value: The value to add to the buffer
+     */
+    public func append<T : Scalar>(value : T){
         let c = MemoryLayout.stride(ofValue: value)
         if c > 8 {
             align(size: 8, additionalBytes: c)
         } else {
             align(size: c, additionalBytes: 0)
         }
         
-        increaseCapacity(size: c)
+        reserveAdditionalCapacity(size: c)
         
         _data.storeBytes(of: value, toByteOffset: leftCursor-c, as: T.self)
         cursor += c
     }
     
+    /**
+     Make offset relative and append it to the buffer
+     
+     - parameters:
+         - offset: The offset to transform and add to the buffer
+     */
     @discardableResult
-    public func putOffset(offset : Offset?) throws -> Int { // make offset relative and put it into byte buffer
+    public func append(offset : Offset?) throws -> Int {
         guard let offset = offset else {
-            put(value: Offset(0))
+            append(value: Offset(0))
             return cursor
         }
         guard offset <= Int32(cursor) else {
-            throw FBBuildError.OffsetIsTooBig
+            throw FlatBuffersBuildError.offsetIsTooBig
         }
         
         if offset == Int32(0) {
-            put(value: Offset(0))
+            append(value: Offset(0))
             return cursor
         }
         align(size: 4, additionalBytes: 0)
         let _offset = Int32(cursor) - offset + MemoryLayout<Int32>.stride;
-        put(value: _offset)
+        append(value: _offset)
         return cursor
     }
-    
-    public func replaceOffset(offset : Offset, atCursor jumpCursor: Int) throws{
+  
+    /**
+     Update an offset in place
+     
+     - parameters:
+         - offset: The new offset to transform and add to the buffer
+         - atCursor: The position to put the new offset to
+     */
+    public func update(offset : Offset, atCursor jumpCursor: Int) throws{
         guard offset <= Int32(cursor) else {
-            throw FBBuildError.OffsetIsTooBig
+            throw FlatBuffersBuildError.offsetIsTooBig
         }
         guard jumpCursor <= cursor else {
-            throw FBBuildError.CursorIsInvalid
+            throw FlatBuffersBuildError.cursorIsInvalid
         }
         let _offset = Int32(jumpCursor) - offset;
         
         _data.storeBytes(of: _offset, toByteOffset: capacity - jumpCursor, as: Int32.self)
     }
-    
-    private func put<T : Scalar>(value : T, at index : Int) {
+ 
+    /**
+     Update a scalar in place
+     
+     - parameters:
+         - value: The new value 
+         - index: The position to modify
+     */
+    private func update<T : Scalar>(value : T, at index : Int) {
         _data.storeBytes(of: value, toByteOffset: index + leftCursor, as: T.self)
     }
     
-    public func openObject(numOfProperties : Int) throws {
+    /**
+     Start an object update sequence
```

Comment:
Not sure about update. I guess Start object construction sequence, is a bit more appropriate.